library/core/src/hint.rs

```diff
@@ -5,27 +5,71 @@
 
 use crate::intrinsics;
 
-/// Informs the compiler that this point in the code is not reachable, enabling
-/// further optimizations.
+/// Informs the compiler that the site which is calling this function is not
+/// reachable, possibly enabling further optimizations.
 ///
 /// # Safety
 ///
-/// Reaching this function is completely *undefined behavior* (UB). In
-/// particular, the compiler assumes that all UB must never happen, and
-/// therefore will eliminate all branches that reach to a call to
+/// Reaching this function is *Undefined Behavior* (UB). In particular, as the
+/// compiler assumes that all forms of Undefined Behavior can never happen, it
+/// will eliminate all branches which themselves reach a call to
```

Comment:
nit: like you said "possibly enabling" above, I think this should also update to
```suggestion
/// may eliminate all branches which themselves reach a call to
```

Reply:
You are referring to "possibly" -> "it will possibly eliminate"? I suggest keeping using strong language here because the programmer should always assume that the compiler *does* eliminate those branches, even if it technically doesn't, yet does other funky stuff which causes time travel and nose daemons.

Maybe more clear: "it will eliminate all branches in the surrounding code which the compiler can determine will invariably lead to a call to `unreachable_unchecked!`."